libcudacxx/include/cuda/std/numbers

```diff
@@ -0,0 +1,383 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDA_STD_NUMBERS
+#define _CUDA_STD_NUMBERS
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__concepts/concept_macros.h>
+#include <cuda/std/__type_traits/enable_if.h>
+#include <cuda/std/__type_traits/is_floating_point.h>
+#include <cuda/std/version>
+
+#if defined(_LIBCUDACXX_HAS_NVFP16)
+#  include <cuda_fp16.h>
+#endif // _LIBCUDACXX_HAS_NVFP16
+#if defined(_LIBCUDACXX_HAS_NVBF16)
+_CCCL_DIAG_PUSH
+_CCCL_DIAG_SUPPRESS_CLANG("-Wunused-function")
+#  include <cuda_bf16.h>
+_CCCL_DIAG_POP
+#endif // _LIBCUDACXX_HAS_NVBF16
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+namespace numbers
+{
+
+template <class _Tp, class = void>
+struct __e
+{};
+
```

Comment:
I believe we also need to enforce https://eel.is/c++draft/numbers#math.constants-3

For that it is the best to have it derive from a struct with an always triggering static_assert
```suggestion

template<class _Tp>
_CCCL_INLINE_VAR __always_false = false;

template <class _Tp>
struct _Ill_formed {
  static_assert(
      __always_false<_Tp>,
      "[math.constants] A program that instantiates a primary template of a mathematical constant variable template is ill-formed.");
};

template <class _Tp, class = void>
struct __e
{};

```